src/renderer/wg_engine/tvgWgGeometry.cpp

```diff
@@ -458,7 +459,8 @@ void WgGeometryData::appendStroke(const WgPolyline* polyline, const RenderStroke
                     } else if (stroke->join == StrokeJoin::Miter) {
                         WgPoint nrm = (nrm0 + nrm1).normal();
                         float cosine = nrm.dot(nrm0);
-                        if ((cosine != 0.0f) && (abs(cosine) < 1.0f) && (abs(wdt / cosine) <= stroke->miterlimit * 2)) {
+                        float mitterRation = 1.0f / std::sin((std::acos(dir0.dot(dir1.negative()))/2.0f));
```

Comment:
this is not optimal that the same logic has to be copied in some other place. but this comment is a  bit wider than this pr
but please apply the same commits as above